clippy_lints/src/methods/mod.rs

```diff
@@ -271,6 +272,43 @@ declare_clippy_lint! {
     "using combinations of `flatten` and `map` which can usually be written as a single method call"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for usage of `_.filter_map(_).map(_)`, which can be more concisely
+    /// written as `_.filter_map(_)`.
+    ///
+    /// **Why is this bad?** Readability, this can be written more concisely as a
+    /// single method call.
+    ///
+    /// ** Example:**
+    /// let ws: Vec<String> = ["is", "this", "shizzle", "for", "rizzle"]
```

Comment:
`rustfmt` suggests this formatting for the example: (Also the backticks are missing)

```suggestion
    /// ```rust
    /// let ws: Vec<String> = ["is", "this", "shizzle", "for", "rizzle"]
    ///     .iter()
    ///     .map(|s| s.to_string())
    ///     .collect();
    ///
    /// let correct: Vec<String> = ws
    ///     .iter()
    ///     .filter_map(|s| if s.contains("izzle") { Some(s) } else { None })
    ///     .map(|s| format!("{}{}", s, s))
    ///     .collect();
    ///
    /// let more_correct: Vec<String> = ws
    ///     .iter()
    ///     .filter_map(|s| {
    ///         if s.contains("izzle") {
    ///             Some(format!("{}{}", s, s))
    ///         } else {
    ///             None
    ///         }
    ///     })
    ///     .collect();
    ///
    /// println!("{:?}", correct);
    /// assert_eq!(correct, more_correct);
    /// ```
```

Reply:
Hmm, I did run `rustfmt` but it didn't make that suggestion for me :man_shrugging: